x-pack/legacy/plugins/uptime/public/state/reducers/monitor.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  MonitorActionTypes,
+  MonitorDetailsState,
+  FETCH_MONITOR_DETAILS,
+  FETCH_MONITOR_DETAILS_SUCCESS,
+  FETCH_MONITOR_DETAILS_FAIL,
+} from '../actions/monitor';
+
+export interface MonitorState {
+  monitorDetailsList: MonitorDetailsState[];
+  loading: boolean;
+  errors: any[];
+}
+
+const initialState: MonitorState = {
+  monitorDetailsList: [],
+  loading: false,
+  errors: [],
+};
+
+export function monitorReducer(state = initialState, action: MonitorActionTypes): MonitorState {
+  switch (action.type) {
+    case FETCH_MONITOR_DETAILS:
+      return {
+        ...state,
+        loading: true,
+      };
+    case FETCH_MONITOR_DETAILS_SUCCESS:
+      const monitorDetails = action.payload;
+      const { monitorId } = monitorDetails;
+      return {
+        ...state,
+        monitorDetailsList: {
+          ...state.monitorDetailsList,
+          [monitorId]: monitorDetails,
```

Comment:
This is pretty nitpicky on my part, but the reduced number of symbols makes this `case` more readable to my eyes. Feel free to ignore this cluster of suggestions.

Reply:
agreed, done !!